tendermint/src/block.rs

```diff
@@ -35,18 +35,18 @@ use tendermint_proto::Protobuf;
 // Default serialization - all fields serialize; used by /block endpoint
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
 #[non_exhaustive]
+#[serde(try_from = "RawBlock", into = "RawBlock")]
 pub struct Block {
     /// Block header
     pub header: Header,
 
     /// Transaction data
-    pub data: transaction::Data,
+    pub data: Vec<Vec<u8>>,
```

Comment:
What's the reasoning behind the change from `Data` (which is `Vec<u8>`) to `Vec<Vec<u8>>`?

Reply:
`Data` isn't a `Vec<u8>`, it holds an `Option<Vec<Transaction>`, where `Transaction` is an opaque wrapper around `Vec<u8>` that blocks using the normal `Vec` methods.  The transaction data is a `Vec<Vec<u8>>` because it's a list of byte sequences, and IMO it's better to use ordinary Rust types that provide convenient methods, unless there is actually some invariant to maintain.

Reply:
Sorry, I was looking at the wrong `Data` struct for some reason ([this one](https://github.com/informalsystems/tendermint-rs/blob/60e1a0a35db90de3de75bfe11e9a2445a7e22431/tendermint/src/abci/data.rs#L10)).

This is something that I've been wondering about for some time now more generally: whether our `Height`/`Round`/etc. types actually add much value beyond just using native Rust types like `u64`/`u32`/etc.

I personally like the simplicity of using Rust native types as far as possible.

Reply:
`Height` used to enforce an invariant (NonZeroU64) but it was changed to allow 0 too at which point having a newtype for it provided little value

Reply:
Having newtypes is really useful for enforcing semantic consistency (am I accidentally using a round in place of a height?), but I think the important thing is that if there aren't any additional invariants on the underlying type, the newtype should have public inner fields, so that people can use all of the existing API.